src/nwscript/decompiler.cpp

```diff
@@ -301,6 +301,11 @@ void Decompiler::writeInstruction(Common::WriteStream &out, const Instruction* i
 		}
 
 		case kOpcodeRSADD: {
+			if (instruction->variables.empty()) {
+				out.writeString("// RSADD not interpretable");
+				break;
+			}
```

Comment:
Wait, how does this even happen?

Reply:
I discovered this in some of the scripts _start functions. A good example is  c_local_set from kotor2

Reply:
The first `RSADDI`?

Hmm, that's the script return value, the `int` in the `int StartingConditional()` xoreos-tools should probably rather create a variable for that.

Reply:
Hmm, wait, or is it?

Reply:
Oh. Looking at it, it seems like yeah, the _start of the script itself will have a return value, not just the main, if it's a StartingConditional.

I.e. that's probably something the stack analysis can do at the last step (or close to). If it detected the script is dialogue conditional script with a StartingConditional (which it already can do), look for the first instruction in _start, and if it's an RSADDI (which it always should be), add a variable there.

But why is it not dectecting that the StartingConditional is copying its return value into there?

@Nostritius, do you feel up to looking into that? I know the script disassembly stuff is a bit of a confusing maze

Reply:
Yeah, it's something I was hoping to not to do something on it. I can take a look on it, but this will probably take some time. In the meantime I would suggest to change the inserted comment into something like: "// TODO: Add analyzation of RSADD in start".